### PR TITLE
PYTHON-2245 Change default uuidRepresentation to UNSPECIFIED (ie disable UUID encoding by default)

### PR DESCRIPTION
**Pull request by ShaneHarvey**

### CODE REVIEW

File: doc/examples/uuid.rst [hunk 1]
```diff
@@ -288,39 +287,117 @@ Supported UUID Representations
...
+``UNSPECIFIED``
+^^^^^^^^^^^^^^^
+
+.. attention:: Starting in PyMongo 4.0,
+   :data:`~bson.binary.UuidRepresentation.UNSPECIFIED` is the default
+   UUID representation used by PyMongo.
+
+The :data:`~bson.binary.UuidRepresentation.UNSPECIFIED` representation
+prevents the incorrect interpretation of UUID bytes by stopping short of
+automatically converting UUID fields in BSON to native UUID types. Loading
```
Comment (behackett) [inline doc/examples/uuid.rst:334]: Loading -> Decoding?

**Review by behackett — COMMENTED** [review]
What about changes to json_util.py? Don't the docs for JSONOptions have to be updated at least?

**Comment by ShaneHarvey** [comment]
> What about changes to json_util.py? Don't the docs for JSONOptions have to be updated at least?

JSONOptions inherits the default uuid_representation from CodecOptions. This is the new default behavior:
```python
>>> json_util.dumps({'uuid': uuid.uuid4()})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/shane/git/mongo-python-driver/bson/json_util.py", line 414, in dumps
    return json.dumps(_json_convert(obj, json_options), *args, **kwargs)
  File "/Users/shane/git/mongo-python-driver/bson/json_util.py", line 449, in _json_convert
    return SON(((k, _json_convert(v, json_options))
  File "/Users/shane/git/mongo-python-driver/bson/son.py", line 42, in __init__
    self.update(data)
  File "/Users/shane/git/mongo-python-driver/bson/son.py", line 145, in update
    for k, v in other:
  File "/Users/shane/git/mongo-python-driver/bson/json_util.py", line 449, in <genexpr>
    return SON(((k, _json_convert(v, json_options))
  File "/Users/shane/git/mongo-python-driver/bson/json_util.py", line 454, in _json_convert
    return default(obj, json_options)
  File "/Users/shane/git/mongo-python-driver/bson/json_util.py", line 842, in default
    binval = Binary.from_uuid(
  File "/Users/shane/git/mongo-python-driver/bson/binary.py", line 245, in from_uuid
    raise ValueError(
ValueError: cannot encode native uuid.UUID with UuidRepresentation.UNSPECIFIED. UUIDs can be manually converted to bson.Binary instances using bson.Binary.from_uuid() or a different UuidRepresentation can be configured. See the documentation for UuidRepresentation for more information.
```

I've updated the docs for CodecOptions, JSONOptions, and MongoClient.

**Review by behackett — APPROVED** [review]

**Review by juliusgeo — APPROVED** [review]
LGTM!

**Pull request merged**